mg/builtins.go

```diff
@@ -31,14 +32,19 @@ func (bcl BultinCmdList) Lookup(name string) (cmd BultinCmd, found bool) {
 	panic("internal error: the `.exec` BuiltinCmd is not defined")
 }
 
+// BuiltinCmds is used to add a list of predefined commands to the state. It
```

Comment:
I'd say something like: `BuiltinCmds implements various builtin commands`. Once you understand how reducers and the state works, it should all be clear how this works so no need to mention it.

Although saying that, it's probably better to just not export this type. `Builtins` is meant to be used like a namespace (so TypeCmd, etc. should be methods instead of global functins) and the type isn't/shouldn't be useful to users outside the package.